hazelcast/protocol/client_message.py

```diff
@@ -223,6 +233,22 @@ def __str__(self):
                                          self.is_flag_set(LISTENER_FLAG),
                                          self.get_data_offset())
 
-    def append_tuple(self, entry_tuple):
-        self.append_data(entry_tuple[0]).append_data(entry_tuple[1])
-        return self
+
+class ClientMessageBuilder(object):
+    def __init__(self, message_callback):
+        self._incomplete_messages = dict()
+        self._message_callback = message_callback
+
+    def on_message(self, client_message):
+        if client_message.is_flag_set(BEGIN_END_FLAG):
+            # handle message
+            self._message_callback(client_message)
+        elif client_message.is_flag_set(BEGIN_FLAG):
+            self._incomplete_messages[client_message.get_correlation_id()] = client_message
+        else:
+            message = self._incomplete_messages[client_message.get_correlation_id()]
```

Comment:
might make sense to add a catch here with a warning, so that it doesn't break reactor thread if it receives a message  which it didn't receive the first part.